gradio/analytics.py

```diff
@@ -116,51 +118,35 @@ def version_check():
         pass
 
 
-def get_local_ip_address() -> str:
-    """
-    Gets the public IP address or returns the string "No internet connection" if unable
-    to obtain it or the string "Analytics disabled" if a user has disabled analytics.
-    Does not make a new request if the IP address has already been obtained in the
-    same Python session.
-    """
-    if not analytics_enabled():
-        return "Analytics disabled"
-
-    if Context.ip_address is None:
-        try:
-            ip_address = httpx.get(
-                "https://checkip.amazonaws.com/", timeout=3
-            ).text.strip()
-        except (httpx.ConnectError, httpx.ReadTimeout):
-            ip_address = "No internet connection"
-        Context.ip_address = ip_address
-    else:
-        ip_address = Context.ip_address
-    return ip_address
+def _write_machine_hash() -> str:
+    machine_hash = secrets.token_urlsafe(32)
+    with open(JSON_PATH, "w+", encoding="utf-8") as j:
```

Comment:
so just to be clear, this function overwrites the entire json file. because we don't store any other information in this file anymore, just the hash? in which case, why even store in json format. just store the hash

Reply:
Yes, storing as json just in case we want to add more fields later on.